src/parser/bison_parser.y

```diff
@@ -689,33 +691,40 @@ select_statement:
 			$$ = $2;
 			$$->withDescriptions = $1;
 		}
-	|	opt_with_clause select_with_paren set_operator select_paren_or_clause opt_order opt_limit {
-			// TODO: allow multiple unions (through linked list)
-			// TODO: capture type of set_operator
-			// TODO: might overwrite order and limit of first select here
+	|	opt_with_clause select_with_paren set_operator select_statement_in_set_operation opt_order opt_limit {
 			$$ = $2;
+			if ($$->setOperators == nullptr) {
+				$$->setOperators = new std::vector<SetOperator*>();
+			}
+			$$->setOperators->push_back($3);
+			$$->setOperators->back()->nestedSelectStatement = $4;
+			$$->setOperators->back()->resultOrder = $5;
+			$$->setOperators->back()->resultLimit = $6;
 			$$->withDescriptions = $1;
-			$$->unionSelect = $4;
-			$$->order = $5;
+		}
+	;
 
-			// Limit could have been set by TOP.
-			if ($6 != nullptr) {
-				delete $$->limit;
-				$$->limit = $6;
-			}
+select_statement_in_set_operation:
+		select_with_paren
+	|	select_part_of_set_operation;
+
+select_part_of_set_operation:
```

Comment:
Terminology: What is the different between a select_statement_in_set_operation and a select_part_of_set_operation?

Reply:
A select_statement_in_set_operation can either have a select statement in parenthesis or one that is not in parenthesis. 

select_part_of_set_operation is the case in which we don't have parenthesis and then we need to make sure that there is no order or limit term compared to a normal statement

Reply:
Mmh, maybe `select_within_set_(operation|definition)` and `select_within_set_definition_no_parentheses`? Also, those two explanations would help as a comment.

Reply:
Seems good to me